### PR TITLE
Task1 ready for review

### PR DESCRIPTION
**Pull request by VasilMitovClouway**
DownloadAgent downloading files.

### CODE REVIEW

File: .idea/uiDesigner.xml [hunk 1]
```diff
@@ -0,0 +1,124 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (freekman) [inline .idea/uiDesigner.xml]: .

File: .idea/modules.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (freekman) [inline .idea/modules.xml]: .

File: .idea/checkstyle-idea.xml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (freekman) [inline .idea/checkstyle-idea.xml]: every time

**Comment by freekman** [comment]
why are the pictures not in the test folder

File: src/com/clouway/DownloadAgent.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class DownloadAgent {
+  private ProgressBar progressBar;
+
+  /**
+   * Constructor
+   * @param progressBar visual feedback for progress.
```
Comment (freekman) [inline src/com/clouway/DownloadAgent.java]: i would like you to read and tell me why are these comments bad
Read from the book ```Clean code :Chapter for the comments```

File: src/com/clouway/DownloadAgent.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  /**
+   * Downloads a file.
+   * @param url for the file
+   * @param destinationFile the place to save the file
+   * @return the donwloaded file
+   * @throws IOException
+   */
+  public File downloadFile(String url, String destinationFile) throws IOException {
```
Comment (freekman) [inline src/com/clouway/DownloadAgent.java]: just ```download```

File: src/com/clouway/DownloadAgent.java [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    URL urlForFile = new URL(url);
+    URLConnection urlConnection = new URL(url).openConnection();
+    Integer fileSize = urlConnection.getContentLength();
+    InputStream inputStream = new BufferedInputStream(urlForFile.openStream());
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
```
Comment (freekman) [inline src/com/clouway/DownloadAgent.java]: why dont you open a FileOutputStream instead an write directly in it and remove the ```writeFileTo````

File: src/com/clouway/DownloadAgent.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  }
+
+  /**
+   * Write the downloaded file to the destination we desire.
+   * @param name name of the file
+   * @param bytes the byte buffer where the file is.
+   * @return the downloaded file.
+   * @throws IOException
+   */
+  private File writeFileTo(String name, byte[] bytes) throws IOException {
```
Comment (freekman) [inline src/com/clouway/DownloadAgent.java]: bad name

File: src/com/clouway/ProgressBar.java [hunk 8]
```diff
@@ -0,0 +1,8 @@
+package com.clouway;
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public interface ProgressBar {
+  Integer downloadProgress(Integer progress);
```
Comment (freekman) [inline src/com/clouway/ProgressBar.java]: ProgressBar.downloadProgress does not sound good why not ProgressBar.update

File: test/com/clouway/DownloadAgentTest.java [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public class DownloadAgentTest {
+  JUnitRuleMockery context = new JUnitRuleMockery();
+  ProgressBar progressBar = context.mock(ProgressBar.class);
+
+  @Test
+  public void downloadingFile() throws Exception {
```
Comment (freekman) [inline test/com/clouway/DownloadAgentTest.java]: rename  to happyPath()

File: test/com/clouway/DownloadAgentTest.java [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+      allowing(progressBar).downloadProgress(50);
+      allowing(progressBar).downloadProgress(60);
+      allowing(progressBar).downloadProgress(70);
+      allowing(progressBar).downloadProgress(80);
+      allowing(progressBar).downloadProgress(90);
+      allowing(progressBar).downloadProgress(100);
+
+    }});
+    File actual = new File("otherfile.jpg");
+    assertThat(downloadAgent.downloadFile("File:/home/clouway/Downloads/online-url-bibanews.jpg", "somefilefortest.jpg").getTotalSpace(), is(actual.getTotalSpace()));
```
Comment (freekman) [inline test/com/clouway/DownloadAgentTest.java]: make it more readable

```js
assertThat(actual, is(expected))
```

Comment (freekman) [inline test/com/clouway/DownloadAgentTest.java]: for the comparison you could use comething like 

```js
File file1 = new File("file1.txt");
File file2 = new File("file2.txt");
boolean isTwoEqual = FileUtils.contentEquals(file1, file2);
```

File: src/com/clouway/downloadagent/ProgressBar.java [hunk 11]
```diff
@@ -0,0 +1,8 @@
+package com.clouway.downloadagent;
+
+/**
+ * @author Vasil Mitov <v.mitov.clouway@gmail.com>
+ */
+public interface ProgressBar {
+  Integer update(Integer progress);
```
Comment (freekman) [inline src/com/clouway/downloadagent/ProgressBar.java]: make it void

File: src/com/clouway/downloadagent/DownloadAgent.java [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+  /**
+   * Downloads a file.
+   * @param url for the file
+   * @param destinationFile the place to save the file
+   * @return the donwloaded file
+   * @throws IOException
+   */
+  public File downloadFile(String url, String destinationFile) throws IOException {
+    URL urlForFile = new URL(url);
```
Comment (freekman) [inline src/com/clouway/downloadagent/DownloadAgent.java]: rename to just url

File: src/com/clouway/downloadagent/DownloadAgent.java [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+    URLConnection urlConnection = new URL(url).openConnection();
+    Integer fileSize = urlConnection.getContentLength();
+    InputStream inputStream = new BufferedInputStream(urlForFile.openStream());
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
```
Comment (freekman) [inline src/com/clouway/downloadagent/DownloadAgent.java]: this is unnecessary

File: src/com/clouway/downloadagent/DownloadAgent.java [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+public class DownloadAgent {
+  private ProgressBar progressBar;
+
+
+  public DownloadAgent(ProgressBar progressBar) {
+    this.progressBar = progressBar;
+  }
+
+  /**
+   * Downloads a file.
```
Comment (freekman) [inline src/com/clouway/downloadagent/DownloadAgent.java]: .

File: src/com/clouway/downloadagent/DownloadAgent.java (hunk 12 above)
Comment (VasilMitovClouway) [inline src/com/clouway/downloadagent/DownloadAgent.java]: Then i have to change the String url to something else.

File: test/com/clouway/downloadagent/DownloadAgentTest.java [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+      allowing(progressBar).update(40);
+      allowing(progressBar).update(50);
+      allowing(progressBar).update(60);
+      allowing(progressBar).update(70);
+      allowing(progressBar).update(80);
+      allowing(progressBar).update(90);
+      allowing(progressBar).update(100);
+
+    }});
+    File actual=downloadAgent.downloadFile("File:/home/clouway/Downloads/online-url-bibanews.jpg", "somefilefortest.jpg");
```
Comment (freekman) [inline test/com/clouway/downloadagent/DownloadAgentTest.java]: ok let me put it this way when i told you to put the picture in the test folder.
If you merge this and i get the code i wont have the picture and the test will fail

**Comment by freekman** [comment]
LGTM

**Pull request merged**